File: app/src/main/java/com/example/doracachesample/RealTimeRepository.java
package com.example.doracachesample;

import android.content.Context;

import com.example.doracachesample.realtime.RealTimeModel;

import org.jetbrains.annotations.NotNull;

import dora.cache.repository.DoraDatabaseCacheRepository;
import dora.cache.repository.Repository;
import dora.http.DoraCallback;
import dora.http.retrofit.DoraRetrofitManager;

@Repository(modelClass = RealTimeModel.class, isListMode = false)
public class RealTimeRepository extends DoraDatabaseCacheRepository<RealTimeModel> {

    public RealTimeRepository(@NotNull Context context) {
        super(context);
    }

    @Override
    protected void onLoadFromNetwork(@NotNull DoraCallback<RealTimeModel> callback) {
        DoraRetrofitManager.INSTANCE.getService(WeatherService.class).getRealTime().enqueue(callback);
    }

    @Override
    public boolean isNoNetworkMode() {
        return false;
    }
}

File: app/src/main/java/com/example/doracachesample/repository/RealTimeRepository.kt
package com.example.doracachesample.repository

import android.content.Context
import com.example.doracachesample.biz.weather.WeatherService
import com.example.doracachesample.model.RealTimeModel
import dora.cache.repository.DoraDatabaseCacheRepository
import dora.cache.repository.Repository
import dora.http.DoraCallback
import dora.http.retrofit.DoraRetrofitManager

@Repository(isListMode = false)
class RealTimeRepository(context: Context) : DoraDatabaseCacheRepository<RealTimeModel>(context) {

    var latlng: String = ""

    override fun onLoadFromNetwork(callback: DoraCallback<RealTimeModel>) {
        DoraRetrofitManager.getService(WeatherService::class.java).getRealTime(latlng).enqueue(callback)
    }
}